db_stress_tool/db_stress_test_base.cc

```diff
@@ -103,12 +103,16 @@ StressTest::StressTest()
   }
 }
 
-StressTest::~StressTest() {
+void StressTest::CleanUp() {
   for (auto cf : column_families_) {
     delete cf;
   }
   column_families_.clear();
+  if (db_) {
```

Comment:
> MultiOpsTxnsStressListener::OnCompactionCompleted() access db_ ...

Will this happen when we are trying to `if (db_) {}..` here in the CleanUp()?

Reply:
It can happen, I think during Close() we will wait for compactions to finish and stop further compactions.

Reply:
It's interesting that the listener only reads the db pointer to check the DB has not been closed yet 

```
  void OnCompactionCompleted(DB* db, const CompactionJobInfo& info) override {
    assert(db);
#ifdef NDEBUG
    (void)db;
#endif
    assert(info.cf_id == 0);
    const ReadOptions read_options(Env::IOActivity::kCompaction);
    stress_test_->VerifyPkSkFast(read_options, info.job_id);
  }
```

Reply:
There is more usage of db_ inside VerifyPkSkFast() below.